specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2017-06-01/Product.json

```diff
@@ -143,6 +143,147 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products/_all/GetProducts": {
+      "post": {
+        "x-ms-examples": {
+          "Returns a list of products.": {
+            "$ref": "examples/Product/ListPost.json"
+          }
+        },
+        "tags": [
+          "Product"
+        ],
+        "description": "Returns a list of products.",
+        "operationId": "GetProducts_List",
+        "parameters": [
+          {
+            "$ref": "AzureStack.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/RegistrationNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DeviceConfigurationParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ProductList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "AzureStack.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products/{productName}/GetProduct": {
+      "post": {
+        "x-ms-examples": {
+          "Returns the specified product.": {
+            "$ref": "examples/Product/GetPost.json"
+          }
+        },
+        "tags": [
+          "Product"
+        ],
+        "description": "Returns the specified product.",
+        "operationId": "GetProduct",
```

Comment:
rename to Products_GetProduct?

Reply:
If I rename it to Products_GetProduct - i receive a couple more warnings by "autorest" tool. It actually forces this name if I want to avoid warnings.

I still can rename and live with the warnings, though, Should I?

Reply:
`Products_Get`

Reply:
Products_get is already taken for the GET api . This is post call. I think you want to rename it to Products_GetProduct.. I think autorest SDK generation tool expects the form of OperationGroup_Operation @jhendrixMSFT can you pls confirm?

Reply:
If you name them both `Products_GetProduct` the code generators will apply their own rules to disambiguate the names which is undesirable.  I agree this one should be `Products_GetProduct`, let's rename the other one.

Reply:
renamed to Products_GetProduct